src/history/history.go

```diff
@@ -202,3 +202,8 @@ func subdomainAvailable(subdomain string) bool {
 	}
 	return true
 }
+
+// GetHistory - Returns the history instance
+func GetHistory() *map[string]*Service {
```

Comment:
Do not use a direct pointer access. Please delete this method. This is not going to be thread safe. Use the `history.Services()` and then `history.SetState()`

Reply:
Also, maps are always returned by reference in golang. See https://stackoverflow.com/questions/40680981/are-maps-passed-by-value-or-by-reference-in-go